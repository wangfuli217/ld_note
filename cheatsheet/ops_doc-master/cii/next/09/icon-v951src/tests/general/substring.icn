#SRC: JCON

# string subscripting test

procedure main()
   local i, j, k, s, t

   s := "abcde"
   t := "ABCDE"
   write("A. ", !s)
   every write("B. ", !s)
   every i := 0 to 6 do write("C ", i, ". ", s[i] | "--")
   every i := 0 to -6 by -1 do write("D ", i, ". ", s[i] | "--")
   every i := -10 to 10 do write("E ", i, ". ", s[3:i] | "--")
   every i := -10 to 5 do write("F ", i, ". ", s[3+:i] | "--") #some SHOULD fail
   every i := -5 to 10 do write("G ", i, ". ", s[3-:i] | "--") #some SHOULD fail

   !s := "X"
   write("H. ", s)
   every !s := "Y"
   write("I. ", s)

   every i := -6 to 6 do {
      s := "abcde"
      if s[i] := t[i] then
         write("J ", i, ". ", s)
      else
         write("J ", i, ". --")
      }

   every i := 1 to 6 do {
      every j := 1 to 6 do {
	 s := "abcde"
	 writes("K ", i, " ", j, ". ")
	 if s[i:j] := "(*)" then
	    write(s)
	 else
	    write(s, " [failed]")
         }
      }

   every i := 1 to 6 do {
      every j := 1 to 6 do {
         every k := 1 to 6 do {
	    s := "abcde"
	    writes("L ", i, " ", j, " ", k, ". ")
	    if s[i:j][k:2] := "(*)" then
	       write(s)
	    else
	       write(s, " [failed]")
	    }
         }
      }

   s := "abcde"
   every !s <- "-" do write("M ", s)
   every s [1 to 5] <- "-" do write("N ", s)
   every s [(-5 to 6) +: 0] <- "--" do write("O ", s)

   s := "abcde"
   every s[2:4] := !"123" do write("P ", s)
   s := "fghij"
   every s[2:4] := !"456" do { write("Q ", s); s := "klmno" }
   
end
