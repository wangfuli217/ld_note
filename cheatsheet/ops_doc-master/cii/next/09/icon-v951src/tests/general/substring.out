A. a
B. a
B. b
B. c
B. d
B. e
C 0. --
C 1. a
C 2. b
C 3. c
C 4. d
C 5. e
C 6. --
D 0. --
D -1. e
D -2. d
D -3. c
D -4. b
D -5. a
D -6. --
E -10. --
E -9. --
E -8. --
E -7. --
E -6. --
E -5. ab
E -4. b
E -3. 
E -2. c
E -1. cd
E 0. cde
E 1. ab
E 2. b
E 3. 
E 4. c
E 5. cd
E 6. cde
E 7. --
E 8. --
E 9. --
E 10. --
F -10. --
F -9. --
F -8. ab
F -7. b
F -6. 
F -5. c
F -4. cd
F -3. cde
F -2. ab
F -1. b
F 0. 
F 1. c
F 2. cd
F 3. cde
F 4. --
F 5. --
G -5. --
G -4. --
G -3. cde
G -2. cd
G -1. c
G 0. 
G 1. b
G 2. ab
G 3. cde
G 4. cd
G 5. c
G 6. 
G 7. b
G 8. ab
G 9. --
G 10. --
H. Xbcde
I. YYYYY
J -6. --
J -5. Abcde
J -4. aBcde
J -3. abCde
J -2. abcDe
J -1. abcdE
J 0. --
J 1. Abcde
J 2. aBcde
J 3. abCde
J 4. abcDe
J 5. abcdE
J 6. --
K 1 1. (*)abcde
K 1 2. (*)bcde
K 1 3. (*)cde
K 1 4. (*)de
K 1 5. (*)e
K 1 6. (*)
K 2 1. (*)bcde
K 2 2. a(*)bcde
K 2 3. a(*)cde
K 2 4. a(*)de
K 2 5. a(*)e
K 2 6. a(*)
K 3 1. (*)cde
K 3 2. a(*)cde
K 3 3. ab(*)cde
K 3 4. ab(*)de
K 3 5. ab(*)e
K 3 6. ab(*)
K 4 1. (*)de
K 4 2. a(*)de
K 4 3. ab(*)de
K 4 4. abc(*)de
K 4 5. abc(*)e
K 4 6. abc(*)
K 5 1. (*)e
K 5 2. a(*)e
K 5 3. ab(*)e
K 5 4. abc(*)e
K 5 5. abcd(*)e
K 5 6. abcd(*)
K 6 1. (*)
K 6 2. a(*)
K 6 3. ab(*)
K 6 4. abc(*)
K 6 5. abcd(*)
K 6 6. abcde(*)
L 1 1 1. abcde [failed]
L 1 1 2. abcde [failed]
L 1 1 3. abcde [failed]
L 1 1 4. abcde [failed]
L 1 1 5. abcde [failed]
L 1 1 6. abcde [failed]
L 1 2 1. (*)bcde
L 1 2 2. a(*)bcde
L 1 2 3. abcde [failed]
L 1 2 4. abcde [failed]
L 1 2 5. abcde [failed]
L 1 2 6. abcde [failed]
L 1 3 1. (*)bcde
L 1 3 2. a(*)bcde
L 1 3 3. a(*)cde
L 1 3 4. abcde [failed]
L 1 3 5. abcde [failed]
L 1 3 6. abcde [failed]
L 1 4 1. (*)bcde
L 1 4 2. a(*)bcde
L 1 4 3. a(*)cde
L 1 4 4. a(*)de
L 1 4 5. abcde [failed]
L 1 4 6. abcde [failed]
L 1 5 1. (*)bcde
L 1 5 2. a(*)bcde
L 1 5 3. a(*)cde
L 1 5 4. a(*)de
L 1 5 5. a(*)e
L 1 5 6. abcde [failed]
L 1 6 1. (*)bcde
L 1 6 2. a(*)bcde
L 1 6 3. a(*)cde
L 1 6 4. a(*)de
L 1 6 5. a(*)e
L 1 6 6. a(*)
L 2 1 1. (*)bcde
L 2 1 2. a(*)bcde
L 2 1 3. abcde [failed]
L 2 1 4. abcde [failed]
L 2 1 5. abcde [failed]
L 2 1 6. abcde [failed]
L 2 2 1. abcde [failed]
L 2 2 2. abcde [failed]
L 2 2 3. abcde [failed]
L 2 2 4. abcde [failed]
L 2 2 5. abcde [failed]
L 2 2 6. abcde [failed]
L 2 3 1. a(*)cde
L 2 3 2. ab(*)cde
L 2 3 3. abcde [failed]
L 2 3 4. abcde [failed]
L 2 3 5. abcde [failed]
L 2 3 6. abcde [failed]
L 2 4 1. a(*)cde
L 2 4 2. ab(*)cde
L 2 4 3. ab(*)de
L 2 4 4. abcde [failed]
L 2 4 5. abcde [failed]
L 2 4 6. abcde [failed]
L 2 5 1. a(*)cde
L 2 5 2. ab(*)cde
L 2 5 3. ab(*)de
L 2 5 4. ab(*)e
L 2 5 5. abcde [failed]
L 2 5 6. abcde [failed]
L 2 6 1. a(*)cde
L 2 6 2. ab(*)cde
L 2 6 3. ab(*)de
L 2 6 4. ab(*)e
L 2 6 5. ab(*)
L 2 6 6. abcde [failed]
L 3 1 1. (*)bcde
L 3 1 2. a(*)bcde
L 3 1 3. a(*)cde
L 3 1 4. abcde [failed]
L 3 1 5. abcde [failed]
L 3 1 6. abcde [failed]
L 3 2 1. a(*)cde
L 3 2 2. ab(*)cde
L 3 2 3. abcde [failed]
L 3 2 4. abcde [failed]
L 3 2 5. abcde [failed]
L 3 2 6. abcde [failed]
L 3 3 1. abcde [failed]
L 3 3 2. abcde [failed]
L 3 3 3. abcde [failed]
L 3 3 4. abcde [failed]
L 3 3 5. abcde [failed]
L 3 3 6. abcde [failed]
L 3 4 1. ab(*)de
L 3 4 2. abc(*)de
L 3 4 3. abcde [failed]
L 3 4 4. abcde [failed]
L 3 4 5. abcde [failed]
L 3 4 6. abcde [failed]
L 3 5 1. ab(*)de
L 3 5 2. abc(*)de
L 3 5 3. abc(*)e
L 3 5 4. abcde [failed]
L 3 5 5. abcde [failed]
L 3 5 6. abcde [failed]
L 3 6 1. ab(*)de
L 3 6 2. abc(*)de
L 3 6 3. abc(*)e
L 3 6 4. abc(*)
L 3 6 5. abcde [failed]
L 3 6 6. abcde [failed]
L 4 1 1. (*)bcde
L 4 1 2. a(*)bcde
L 4 1 3. a(*)cde
L 4 1 4. a(*)de
L 4 1 5. abcde [failed]
L 4 1 6. abcde [failed]
L 4 2 1. a(*)cde
L 4 2 2. ab(*)cde
L 4 2 3. ab(*)de
L 4 2 4. abcde [failed]
L 4 2 5. abcde [failed]
L 4 2 6. abcde [failed]
L 4 3 1. ab(*)de
L 4 3 2. abc(*)de
L 4 3 3. abcde [failed]
L 4 3 4. abcde [failed]
L 4 3 5. abcde [failed]
L 4 3 6. abcde [failed]
L 4 4 1. abcde [failed]
L 4 4 2. abcde [failed]
L 4 4 3. abcde [failed]
L 4 4 4. abcde [failed]
L 4 4 5. abcde [failed]
L 4 4 6. abcde [failed]
L 4 5 1. abc(*)e
L 4 5 2. abcd(*)e
L 4 5 3. abcde [failed]
L 4 5 4. abcde [failed]
L 4 5 5. abcde [failed]
L 4 5 6. abcde [failed]
L 4 6 1. abc(*)e
L 4 6 2. abcd(*)e
L 4 6 3. abcd(*)
L 4 6 4. abcde [failed]
L 4 6 5. abcde [failed]
L 4 6 6. abcde [failed]
L 5 1 1. (*)bcde
L 5 1 2. a(*)bcde
L 5 1 3. a(*)cde
L 5 1 4. a(*)de
L 5 1 5. a(*)e
L 5 1 6. abcde [failed]
L 5 2 1. a(*)cde
L 5 2 2. ab(*)cde
L 5 2 3. ab(*)de
L 5 2 4. ab(*)e
L 5 2 5. abcde [failed]
L 5 2 6. abcde [failed]
L 5 3 1. ab(*)de
L 5 3 2. abc(*)de
L 5 3 3. abc(*)e
L 5 3 4. abcde [failed]
L 5 3 5. abcde [failed]
L 5 3 6. abcde [failed]
L 5 4 1. abc(*)e
L 5 4 2. abcd(*)e
L 5 4 3. abcde [failed]
L 5 4 4. abcde [failed]
L 5 4 5. abcde [failed]
L 5 4 6. abcde [failed]
L 5 5 1. abcde [failed]
L 5 5 2. abcde [failed]
L 5 5 3. abcde [failed]
L 5 5 4. abcde [failed]
L 5 5 5. abcde [failed]
L 5 5 6. abcde [failed]
L 5 6 1. abcd(*)
L 5 6 2. abcde(*)
L 5 6 3. abcde [failed]
L 5 6 4. abcde [failed]
L 5 6 5. abcde [failed]
L 5 6 6. abcde [failed]
L 6 1 1. (*)bcde
L 6 1 2. a(*)bcde
L 6 1 3. a(*)cde
L 6 1 4. a(*)de
L 6 1 5. a(*)e
L 6 1 6. a(*)
L 6 2 1. a(*)cde
L 6 2 2. ab(*)cde
L 6 2 3. ab(*)de
L 6 2 4. ab(*)e
L 6 2 5. ab(*)
L 6 2 6. abcde [failed]
L 6 3 1. ab(*)de
L 6 3 2. abc(*)de
L 6 3 3. abc(*)e
L 6 3 4. abc(*)
L 6 3 5. abcde [failed]
L 6 3 6. abcde [failed]
L 6 4 1. abc(*)e
L 6 4 2. abcd(*)e
L 6 4 3. abcd(*)
L 6 4 4. abcde [failed]
L 6 4 5. abcde [failed]
L 6 4 6. abcde [failed]
L 6 5 1. abcd(*)
L 6 5 2. abcde(*)
L 6 5 3. abcde [failed]
L 6 5 4. abcde [failed]
L 6 5 5. abcde [failed]
L 6 5 6. abcde [failed]
L 6 6 1. abcde [failed]
L 6 6 2. abcde [failed]
L 6 6 3. abcde [failed]
L 6 6 4. abcde [failed]
L 6 6 5. abcde [failed]
L 6 6 6. abcde [failed]
M -bcde
M a-cde
M ab-de
M abc-e
M abcd-
N -bcde
N a-cde
N ab-de
N abc-e
N abcd-
O --abcde
O a--bcde
O ab--cde
O abc--de
O abcd--e
O abcde--
O --abcde
O a--bcde
O ab--cde
O abc--de
O abcd--e
O abcde--
P a1de
P a2de
P a3de
Q f4ij
Q k5mno
Q k6mno
